tests/behat/behat-features/src/PushingContext.php

```diff
@@ -0,0 +1,295 @@
+  }
+
+  /**
+   * Login to contentpool on user login page.
+   */
+  protected function loginToContentpool() {
+    // Login as user.
+    $element = $this->getSession()->getPage();
+    $element->fillField($this->getDrupalText('username_field'), 'dru_admin');
+    $element->fillField($this->getDrupalText('password_field'), 'changeme');
+    $submit = $element->findButton($this->getDrupalText('log_in'));
+    if (empty($submit)) {
+      throw new ExpectationException(sprintf("No submit button at %s", $this->getSession()->getCurrentUrl()));
+    }
+    $submit->click();
+    // Quick check that user was logged in successfully.
+    $this->assertSession()->pageTextContains("Member for");
+  }
+
+  /**
+   * I am logged in to contentpool.
+   *
+   * @Given I am logged in to contentpool
+   */
+  public function loggedInToContentpool() {
+    // Visit user login page on contentpool.
+    $this->visitContentpoolPath('user/login');
+    // Login to contentpool.
+    $this->loginToContentpool();
+  }
+
+  /**
+   * I click push notification link for current site.
+   *
+   * @Then I click push notification link for current site
+   */
+  public function clickPushNotificationLinkForCurrentSite() {
+    $site_uuid = \Drupal::config('system.site')->get('uuid');
+    $xpath = "//table//td[text()='$site_uuid']/../td/a[@title='Click to enable'][1]";
+    $push_notification_link_element = $this->getSession()->getPage()->find('xpath', $xpath);
+    if (!$push_notification_link_element) {
+      throw new ExpectationException('Push notification link not found."', $this->getSession());
+    }
+    $push_notification_link = $push_notification_link_element->getAttribute('href');
+    if (!$push_notification_link) {
+      throw new ExpectationException('Push notification link is empty."', $this->getSession());
+    }
+    $this->visitPath($push_notification_link);
+  }
+
+  /**
+   * Fills in form field with a value and random suffix.
+   *
+   * @When I fill in :field with :value and random suffix
+   */
+  public function fillFieldWithValueAndRandomSuffix($field, $value) {
+    $this->randomSuffix = rand(0, 100000);
+    $this->getSession()->getPage()->fillField($field, $value . $this->randomSuffix);
+  }
+
+  /**
+   * Fills in form field with a value and last random suffix.
+   *
+   * @When I fill in :field with :value and last random suffix
+   */
+  public function fillFieldWithValueAndLastRandomSuffix($field, $value) {
+    $this->getSession()->getPage()->fillField($field, $value . $this->randomSuffix);
+  }
+
+  /**
+   * I should see text with last random suffix.
+   *
+   * @Then I should see :text with last random suffix
+   */
+  public function shouldSeeTextWithLastRandomSuffix($text) {
+    $text .= $this->randomSuffix;
+    $this->assertSession()->pageTextContains($text);
+  }
+
+  /**
+   * Wait for the page to be loaded.
+   *
+   * @When I wait for the page to be loaded
+   */
+  public function waitForThePageToBeLoaded() {
```

Comment:
I guess we should move things like that in our behat-utils, but out of scope here. Feel free to do it whenever you need it next of course.